pkg/controller/compute/eks.go

```diff
@@ -273,26 +266,20 @@ func (r *Reconciler) _awsauth(cluster *eks.Cluster, instance *awscomputev1alpha3
 	return err
 }
 
-func (r *Reconciler) _sync(instance *awscomputev1alpha3.EKSCluster, client eks.Client) (reconcile.Result, error) {
-	cluster, err := client.Get(instance.Status.ClusterName)
-	if err != nil {
-		return r.fail(instance, err)
-	}
-
+func (r *Reconciler) _sync(instance *awscomputev1alpha3.EKSCluster, cluster *eks.Cluster, client eks.Client) (reconcile.Result, error) {
```

Comment:
Would it make sense to just pass the version here rather than the whole `*eks.Cluster`, since that's the only thing we use?

Reply:
Not sure if I'm looking at the place you're referring to. In `_sync` function, `cluster` is used by [`r.secret`](https://github.com/crossplane/provider-aws/pull/184/files#diff-cd0d23278d543d908fd906517f19a18dR310) and [`r.auth`](https://github.com/crossplane/provider-aws/pull/184/files#diff-cd0d23278d543d908fd906517f19a18dR306) calls as well as for its other properties like [`endpoint`](https://github.com/crossplane/provider-aws/pull/184/files#diff-cd0d23278d543d908fd906517f19a18dR315) and state.